src/main/java/me/xemor/userinterface/UserInterface.java

```diff
@@ -4,33 +4,38 @@
 import net.kyori.adventure.platform.bukkit.BukkitAudiences;
 import org.bukkit.plugin.java.JavaPlugin;
 
-public final class UserInterface extends JavaPlugin {
-
-    private static UserInterface userInterface;
-    private static boolean hasFloodgate;
+@SuppressWarnings("unused")
+public final class UserInterface {
+    private static JavaPlugin plugin;
     private static BukkitAudiences bukkitAudiences;
+    private static boolean hasFloodgate;
+
+    public static void enable(JavaPlugin plugin) {
+        UserInterface.plugin = plugin;
 
-    @Override
-    public void onEnable() {
-        // Plugin startup logic
-        userInterface = this;
-        bukkitAudiences = BukkitAudiences.create(this);
+        bukkitAudiences = BukkitAudiences.create(plugin);
         ChestHandler chestHandler = new ChestHandler();
 
-        if (this.getServer().getPluginManager().getPlugin("Floodgate") != null) hasFloodgate= true;
-        else {
+        if (plugin.getServer().getPluginManager().getPlugin("Floodgate") != null) {
+            hasFloodgate = true;
+            plugin.getLogger().info("Floodgate plugin found. Enabling Floodgate support.");
```

Comment:
tbh I would either make this silent or make it explicitly reference the UserInterfaces dependency

Reply:
Ohh very true